src/api/apis/OMDbAPI.ts

```diff
@@ -29,7 +29,8 @@ export class OMDbAPI extends APIModel {
 		console.log(`MDB | api "${this.apiName}" queried by Title`);
 
 		if (!this.plugin.settings.OMDbKey) {
-			throw Error(`MDB | API key for ${this.apiName} missing.`);
+			console.error(Error(`MDB | API key for ${this.apiName} missing.`));
```

Comment:
The change for getById is missing.

Reply:
Actually I left the getById as it was because it is supposed to fail when it's missing that specific API key. I can't have it return an empty array since it would give an error (Type 'undefined[]' is missing the following properties from type 'MediaTypeModel': type, subType, title, englishTitle, and 10 more).

Ideally rather than silently fail it would give the user a message box saying why it failed.
Originally this PR was intended as just a bug fix for video game searches but I don't mind expanding it to also adding a visible error message. Would it be alright to just make a new Modal for warnings/errors based on the PreviewModal?

Reply:
> Would it be alright to just make a new Modal for warnings/errors based on the PreviewModal?

I would use Obsidians `Notice` to give a small push notification. An entire modal seems overkill.

Reply:
Oh wow I never knew about that function, that was very easy to add.